src/Pyz/Yves/ProductPackagingUnitWidget/Theme/default/components/molecules/packaging-unit-quantity-selector/packaging-unit-quantity-selector.ts

```diff
@@ -1,806 +1,91 @@
-                if (salesUnitId == this.leadSalesUnits[key].id_product_measurement_sales_unit) {
-                    return this.leadSalesUnits[key];
-                }
-            }
-        }
-    }
-
-    private getMinAmount() {
-        if (
-            typeof this.productPackagingUnitStorage !== 'undefined' &&
-            this.productPackagingUnitStorage.hasOwnProperty('amount_min') &&
-            this.productPackagingUnitStorage.amount_min !== null
-        ) {
-            return Number(this.productPackagingUnitStorage.amount_min);
-        }
-
-        return 1;
-    }
-
-    private getMaxAmount() {
-        if (
-            typeof this.productPackagingUnitStorage !== 'undefined' &&
-            this.productPackagingUnitStorage.hasOwnProperty('amount_max') &&
-            this.productPackagingUnitStorage.amount_max !== null
-        ) {
-            return Number(this.productPackagingUnitStorage.amount_max);
-        }
-
-        return 0;
-    }
-
-    private getAmountInterval() {
-        if (
-            typeof this.productPackagingUnitStorage !== 'undefined' &&
-            this.productPackagingUnitStorage.hasOwnProperty('amount_interval') &&
-            this.productPackagingUnitStorage.amount_interval !== null
-        ) {
-            return Number(this.productPackagingUnitStorage.amount_interval);
-        }
-
-        return 1;
-    }
-
-    private getMinAmountChoice(amountInSalesUnits: number) {
-        const amountInBaseUnits = Number(
-            (
-                (amountInSalesUnits * this.precision * Number(this.currentLeadSalesUnit.conversion)) /
-                this.precision
-            ).toFixed(this.numberOfDecimalPlaces),
-        );
-
-        if (amountInBaseUnits < this.getMinAmount()) {
-            return this.getMinAmount();
-        }
-
-        if (this.isAmountGreaterThanMaxAmount(amountInBaseUnits)) {
-            return 0;
+            this.amountInSalesUnitInput.step = String(amountInSalesUnitsStep);
         }
 
-        if (this.isAmountMultipleToInterval(amountInBaseUnits)) {
-            return this.getMinAmountChoice(
-                (amountInBaseUnits - this.getAmountPercentageOfDivision(amountInBaseUnits)) /
-                    this.currentLeadSalesUnit.conversion,
-            );
-        }
-
-        return amountInBaseUnits;
+        this.onAmountInputChange(amountInSalesUnits);
     }
 
-    private getMaxAmountChoice(amountInSalesUnits: number, minChoice: number) {
-        let amountInBaseUnits = Number(
-            (
-                (amountInSalesUnits * this.precision * Number(this.currentLeadSalesUnit.conversion)) /
-                this.precision
-            ).toFixed(this.numberOfDecimalPlaces),
-        );
-
-        if (this.isAmountGreaterThanMaxAmount(amountInBaseUnits)) {
-            amountInBaseUnits = this.getMaxAmount();
+    protected multiply(a: number, b: number): number {
+        const result = a * b;
+        const precision = 1000;
```

Comment:
Why do you hardcode precision and not from `this.currentSalesUnit.precision` ?
As I see it is also used in the `convertBaseUnitsAmountToCurrentSalesUnitsAmount` method.

Reply:
Core implementation leads to the bug in the demoshop.
Created a ticket to resolve it https://spryker.atlassian.net/browse/CC-29470